client: fix console --loop on microcom error

Bastian-Krause opened the pull request:
**Description**
In certain situation, e.g. if the serial port vanishes (i.e. on-board FTDI chips), microcom returns with 1 and:

```
Got EOF from port
connection lost
```

The loop should still continue, so do not raise an exception in this case.

Fixes: cdebb1cc ("remote/client: return exit code for ssh/scp/rsync/telnet/video/audio/console")

**Checklist**
- [x] PR has been tested

Fixes #1137

Bastian-Krause commented:
Something's not correct, yet. On unlock, this happens:
```
Got EOF from port
connection lost
rlabb-srv None
Traceback (most recent call last):
  File "[...]/labgrid/labgrid/remote/client.py", line 1883, in main
    session.loop.run_until_complete(coro)
  File "/usr/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "[...]/labgrid/labgrid/remote/client.py", line 858, in console
    res = await self._console(place, target, 10.0, logfile=self.args.logfile,
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "[...]/labgrid/labgrid/remote/client.py", line 820, in _console
    assert port is not None, "Port is not set"
           ^^^^^^^^^^^^^^^^
AssertionError: Port is not set
```

jluebbe commented:
There is a typo in the commit title.

khilman reviewed (approved):
Tested-by: Kevin Hilman <khilman@baylibre.com>

Bastian-Krause commented:
> Something's not correct, yet. On unlock, this happens:
> 
> ```
> Got EOF from port
> connection lost
> Traceback (most recent call last):
>   File "[...]/labgrid/labgrid/remote/client.py", line 1883, in main
>     session.loop.run_until_complete(coro)
>   File "/usr/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
>     return future.result()
>            ^^^^^^^^^^^^^^^
>   File "[...]/labgrid/labgrid/remote/client.py", line 858, in console
>     res = await self._console(place, target, 10.0, logfile=self.args.logfile,
>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "[...]/labgrid/labgrid/remote/client.py", line 820, in _console
>     assert port is not None, "Port is not set"
>            ^^^^^^^^^^^^^^^^
> AssertionError: Port is not set
> ```

Fixed by explicitly checking whether the place is no longer acquired.

> There is a typo in the commit title.

Fixed.

Also simplified the logic in `console()`.

Bastian-Krause commented:
@khilman Testing one more time would be much appreciated.

khilman reviewed (approved):
Tested-by: Kevin Hilman <khilman@baylibre.com>
